### PR TITLE
Issue#727 disable deleting

### PR DESCRIPTION
**Pull request by MarselAhmetov**
Added new services, class for storing constants, new Feature (TOPIC_DELETION) and FeatureService getAllowedFeatures instead Features getEnabledFeatures, big refactoring started :)

### CODE REVIEW

**Review by tdavletov — APPROVED** [review]

**Comment by tdavletov** [comment]
Great refactoring! I guess ClusterUtil class is the next candidate.

File: kafka-ui-api/src/main/java/com/provectus/kafka/ui/service/ClusterService.java [hunk 1]
```diff
@@ -247,8 +244,15 @@ public TopicsResponse getTopics(String name, Optional<Integer> page,
         .orElseThrow(ClusterNotFoundException::new);
     var topic = getTopicDetails(clusterName, topicName)
         .orElseThrow(TopicNotFoundException::new);
-    return kafkaService.deleteTopic(cluster, topic.getName())
-        .doOnNext(t -> updateCluster(topicName, clusterName, cluster));
+    return brokerService.getBrokerConfigMap(cluster, cluster.getBrokers().get(0))
```
Comment (germanosin) [inline kafka-ui-api/src/main/java/com/provectus/kafka/ui/service/ClusterService.java:247]: ideally we should check configs on controller node, not first

Comment (germanosin) [reply kafka-ui-api/src/main/java/com/provectus/kafka/ui/service/ClusterService.java:247]: Btw you already have this features in cluster feature list, as for me no sense to recheck it.

File: kafka-ui-api/src/main/java/com/provectus/kafka/ui/service/FeatureServiceImpl.java [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+
+  private boolean topicDeletionCheck(KafkaCluster cluster) {
+    return brokerService.getBrokers(cluster).next()
+        .flatMap(broker -> brokerService.getBrokerConfigMap(cluster, broker.getId()))
+        .map(config -> {
+          if (config != null && config.get(DELETE_TOPIC_ENABLE) != null) {
+            return Boolean.parseBoolean(config.get(DELETE_TOPIC_ENABLE).getValue());
+          }
+          return false;
+        }).blockOptional().orElse(false);
```
Comment (germanosin) [inline kafka-ui-api/src/main/java/com/provectus/kafka/ui/service/FeatureServiceImpl.java:54]: Block looks dirty, could we avoid it?

**Review by germanosin — CHANGES_REQUESTED** [review]

File: kafka-ui-api/src/main/java/com/provectus/kafka/ui/service/FeatureServiceImpl.java (hunk 2 above)
Comment (MarselAhmetov) [reply kafka-ui-api/src/main/java/com/provectus/kafka/ui/service/FeatureServiceImpl.java:54]: I see no way to do this

File: kafka-ui-api/src/main/java/com/provectus/kafka/ui/service/ClusterService.java (hunk 1 above)
Comment (MarselAhmetov) [reply kafka-ui-api/src/main/java/com/provectus/kafka/ui/service/ClusterService.java:247]: It seems to me that we need to have a check not only on the client but also on the server

File: kafka-ui-react-app/src/components/Topics/Topic/Details/Details.tsx [hunk 3]
```diff
@@ -1,7 +1,8 @@
 import React from 'react';
 import { ClusterName, TopicName } from 'redux/interfaces';
-import { Topic, TopicDetails } from 'generated-sources';
+import { ClusterFeaturesEnum, Topic, TopicDetails } from 'generated-sources';
 import { NavLink, Switch, Route, Link, useHistory } from 'react-router-dom';
+import { useSelector } from 'react-redux';
```
Comment (GneyHabub) [inline kafka-ui-react-app/src/components/Topics/Topic/Details/Details.tsx:5]: There's the DetailsContainer component, which is responsible for providing the data from redux to the component. Better use it for the sake of consistency

File: kafka-ui-react-app/src/components/Topics/List/ListItem.tsx [hunk 4]
```diff
@@ -33,6 +36,10 @@ const ListItem: React.FC<ListItemProps> = ({
   clearTopicMessages,
 }) => {
   const { isReadOnly } = React.useContext(ClusterContext);
+  const features = useSelector(getClustersFeatures(clusterName));
```
Comment (workshur) [inline kafka-ui-react-app/src/components/Topics/List/ListItem.tsx:39]: please add this feature to ClusterContext

Comment (workshur) [reply kafka-ui-react-app/src/components/Topics/List/ListItem.tsx:39]: Should be something like `isTopicDeleteAllowed`

File: kafka-ui-react-app/src/components/Topics/Topic/Details/Details.tsx [hunk 5]
```diff
@@ -47,6 +49,11 @@ const Details: React.FC<Props> = ({
     clearTopicMessages(clusterName, topicName);
   }, [clusterName, topicName]);
 
+  const features = useSelector(getClustersFeatures(clusterName));
+  const hasKafkaTopicDeletion = features.includes(
+    ClusterFeaturesEnum.TOPIC_DELETION
+  );
+
```
Comment (workshur) [inline kafka-ui-react-app/src/components/Topics/Topic/Details/Details.tsx:56]: context

**Review by workshur — COMMENTED** [review]
@MBovtriukProvectus Looks good so far. Next time I would suggest you to use own branch/PR

**Review by GneyHabub — APPROVED** [review]

File: kafka-ui-react-app/src/components/Cluster/Cluster.tsx [hunk 6]
```diff
@@ -34,13 +34,18 @@ const Cluster: React.FC = () => {
   const hasSchemaRegistryConfigured = features.includes(
     ClusterFeaturesEnum.SCHEMA_REGISTRY
   );
+  const isTopicDeletionAllowed = features.includes(
+    ClusterFeaturesEnum.TOPIC_DELETION
+  );
   const hasKsqlDbConfigured = features.includes(ClusterFeaturesEnum.KSQL_DB);
 
+
```
Comment (workshur) [inline kafka-ui-react-app/src/components/Cluster/Cluster.tsx:42]: @GneyHabub

**Review by workshur — APPROVED** [review]
LGTM

**Pull request merged**